src/components/swap/SwapButtons/SwapButton.tsx

```diff
@@ -24,23 +24,49 @@ const StyledSwapLoadingButton = styled(ButtonPrimary)`
   background-image: linear-gradient(90deg, #4c4c76 19.74%, #292942 120.26%);
   cursor: 'wait';
   padding: 16px;
+
+  ${({ theme }) => theme.mediaWidth.upToExtraSmall`
+    display: grid;
+    gap: 8px;
+    justify-items: center;
+  `};
 `
 
-const StyledPlataformImage = styled.img`
-  margin-top: -3px;
-  margin-right: 6px;
+const RotatingLogo = styled.div`
+  position: relative;
+  display: flex;
+  justify-content: start;
+  align-items: center;
+  width: 112px;
+  height: 22px;
+
+  ${({ theme }) => theme.mediaWidth.upToExtraSmall`
+    justify-content: center;
+  `};
 `
 
-const StyledSwapButtonText = styled(Text)<{ width?: string }>`
+const StyledSwapButtonText = styled.div<{ width?: string }>`
   display: flex;
-  height: 16px;
-  white-space: pre-wrap;
-  width: ${({ width }) => (width ? `${width}px` : 'auto')};
+  justify-content: center;
+  align-items: center;
+  margin-left: 6px;
```

Comment:
1. `width` is unused so you can remove it.
2. Correct me if I wrong but `flex` is useless here.